crates/bevy_tasks/src/task_pool.rs

```diff
@@ -309,6 +336,42 @@ mod tests {
         assert_eq!(count.load(Ordering::Relaxed), 100);
     }
 
+    #[test]
+    fn test_spawn_pollable() {
+        let transform_fn = |i: usize| i * 3;
+
+        let pool = TaskPool::new();
+        let nums: Vec<usize> = (1..10).into_iter().collect();
+        let pollables: Vec<PollableTask<usize>> = nums
+            .clone()
+            .into_iter()
+            .map(|i| {
+                pool.spawn_pollable(async move {
+                    futures_timer::Delay::new(Duration::from_secs_f32(0.5)).await;
+                    transform_fn(i)
+                })
+            })
+            .collect();
+
+        // Simulate game loop
+        loop {
+            let mut done = true;
+            for i in 0..pollables.len() {
+                let locked = pollables[i].poll().unwrap();
+                if let Some(r) = *locked {
+                    assert_eq!(r, transform_fn(nums[i]));
+                } else {
+                    done = false;
+                    break;
+                }
+            }
+            if done {
+                return;
+            }
+            std::thread::sleep(Duration::from_secs_f32(1.0 / 60.0));
```

Comment:
If you remove the timing based parts of this task, it will make sense to change this to [`std::thread::yield_now`](https://doc.rust-lang.org/std/thread/fn.yield_now.html)